WalletWasabi/WabiSabi/Client/ArenaClient.cs

```diff
@@ -112,30 +112,30 @@ public async Task RemoveInputAsync(uint256 roundId, uint256 aliceId, Cancellatio
 
 		public async Task<ArenaResponse> ReissueCredentialAsync(
 			uint256 roundId,
-			long value1,
-			Script scriptPubKey1,
-			long value2,
-			Script scriptPubKey2,
+			long amount1,
+			int scriptPubKeyVsize1,
+			long amount2,
+			int scriptPubKeyVsize2,
```

Comment:
```suggestion
			int vsize1,
			long amount2,
			int vsize2,
```

these aren't necessarily in correspondence with script sizes, so these names make more sense, and they should probably also be of the same type...

instead of sending two parameters for each, since code does a lot of `new long[] { a, b }`, maybe it's better to make all of these `IEnumerable<long>` or even `IEnumerable<ulong>`?

rationale:

- we use `IEnumerable<long>` in the underlying `WabiSabiClient` API
  - .. but we could use `IEnumerable<ulong>`, internally these `long` values are converted to `ulong` and they are never negative
- `CredentialDependency.Value` is a `ulong`, and you get those by enumerating the `OutEdges` or `InEdges` of a node, so `g.OutEdges(n).Select(x => x.Value)` produces an `IEnumerable<ulong>`
- the dependency graph also uses `IEnumerable<ulong>` and the decomposer produces those

Note that `ArenaClient` and `WabiSabiClient` are pretty inconsistent about this right now,

Reply:
For `IEnumerable<ulong>` I cannot use the `Sum()` method. And there are many more drawbacks which will turn out during the usage for example the underlying libraries are using the or not, otherwise, I will have to cast a lot. The question: is it worth deter from code convention? Unsigned types are exotic types in C# and bring more problems than it usually solves.

Reply:
the largest long value far more exceeds the public const long MaximumNumberOfSatoshis = 2099999997690000; which is also a long.
The rationale of values those cant be negative should be unsigned, does not help, in fact, it has the opposite effect it is dangerous. The default behavior in C# is to swallow arithmetic overflow/underflow so if something underflows it will be unnoticed. With long we can just write if (a < 0)...

Reply:
I guess lack of `Sum` on its own might be a strong enough to prefer `IEnumerable<long>` over `IEnumerable<ulong>`...

That said I don't think it particularly matters which one we use, even if we standardize on `long` it will be a big improvement over the current state of `ArenaClient` in particular where we use like 3 different types for specifying the requested credential values depending on the method (`long`, `Money` and `int`)